querydsl-jpa/src/test/java/com/querydsl/jpa/JPASQLBase.java

```diff
@@ -86,10 +90,49 @@ public void entityQueries_createQuery() {
     @ExcludeIn(Target.MYSQL)
     public void entityQueries_createQuery2() {
         SAnimal cat = new SAnimal("CAT");
-        QCat catEntity = QCat.cat;
 
         Query query = query().from(cat).select(catEntity).createQuery();
         assertEquals(6, query.getResultList().size());
     }
 
+    @Test
+    public void should_fetch_results_with_factory_expression() {
+        final long expectedTotalResultCount = 6;
+        final HashMap<String, Expression<?>> bindings = new HashMap<String, Expression<?>>();
+        bindings.put("name", cat.name);
+
+        final long actualTotalResultCount = query().from(cat)
+                                                   .select(Projections.bean(Cat.class, bindings))
+                                                   .fetchResults()
+                                                   .getTotal();
+
+        assertEquals(expectedTotalResultCount, actualTotalResultCount);
+    }
+
+    @Test
+    public void should_get_grouped_list_by_using_fetch_results() {
+        final long expectedCatColorKindCount = 1;
+
+        final long actualCatColorKindCount = query().from(cat)
+                                                    .select(catEntity.color)
+                                                    .groupBy(catEntity.color)
+                                                    .fetchResults()
+                                                    .getTotal();
+
+        assertEquals(expectedCatColorKindCount, actualCatColorKindCount);
+    }
+
+    @Test
+    public void should_get_black_cat_count_by_using_group_by_and_having() {
+        final long expectedBlackCatCount = 6L;
+
+        final long actualBlackCatCount = query().from(cat)
+                                                .select(catEntity.name.count())
+                                                .groupBy(catEntity.color)
+                                                .having(catEntity.name.count().goe(1))
```

Comment:
This having clause is not entropy reducing, so even if its handled correcly, the query gives the proper result.

Reply:
How should I test it? Can you give me a comment~? 🤔

Reply:
`catEntity.name.count()` can never be 0. It is minimally 1. But it can be higher than 1. Not sure if that occurs in your test data. The point is, `catEntity.name.count() >= 1` will never filter any rows. Therefore you can't know whether the having clause was properly handled. It would be interesting to see what happens for `catEntity.name.count() >= 2`. It should return a non empty result (the ones   with `count() >= 2`), and eliminate the results with `count() = 1`). The latter for example doesn't happen in JPA currently. It doesn't handle the having clause at all. Therefore count queries with a HAVING clause are fundamentally broken in JPQL (not JPA SQL), and it would be interesting to see how this behaves in JPA SQL.

Reply:
Thanks for sharing your thoughts. :)

I have created two additional test data to check the having clause more accurately. (tabbyColorCatFoo, tabbyColorCatBar)

The total test data were 6 black cats and 2 Tabby cats.

And I wrote the query as below.

```
final long actualTabbyCatCount = query().from(cat)
                                                                      .select(catEntity.name.count())
                                                                      .groupBy(catEntity.color)
                                                                      .having(catEntity.name.count().eq(2L))
                                                                      .fetchOne();

assertEquals(expectedTabbyCatCount, actualTabbyCatCount); //true
```